python/ray/test/test_ray_init.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import pytest
+import redis
+
+import ray
+
+
+@pytest.fixture
+def start_ray_with_password():
+    ray.shutdown()
+
+    password = "some_password"
+    exception = None
+
+    # Fix for #3045
+    global f
+    f = ray.remote(f._function)
+
+    try:
+        info = ray.init(redis_password=password)
+    except Exception as e:
+        info = ray.init(redis_password=None)
+        exception = e
+    use_raylet = ray.global_state.use_raylet
+
+    return password, info, exception, use_raylet
+
+
+@pytest.fixture
+def use_credis():
+    return ("RAY_USE_NEW_GCS" in os.environ)
+
+
+@ray.remote
+def f():
+    return 1
+
+
+class TestRedisPassword(object):
+    def test_raylet_only(self, start_ray_with_password, use_credis):
+        password, info, exception, use_raylet = start_ray_with_password
+        if use_raylet and not use_credis:
+            assert exception is None
```

Comment:
It's natural to just run the test and if it raises an exception, then pytest will report the error. Instead, we're catching the exception and then asserting that the exception is not `None`, which is very roundabout.